packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -503,6 +504,86 @@ class ContainerLayer extends Layer {
     return child == equals;
   }
 
+  PictureLayer _highlightConflictingLayer(PhysicalModelLayer child) {
+    final ui.PictureRecorder recorder = ui.PictureRecorder();
+    final Canvas canvas = Canvas(recorder);
+    canvas.drawPath(
+      child.clipPath,
+      Paint()
+        ..color = const Color(0xFFAA0000)
+        ..style = PaintingStyle.stroke
+        ..strokeWidth = child.elevation + 10.0,
+    );
+    final PictureLayer pictureLayer = PictureLayer(child.clipPath.getBounds())
+      ..picture = recorder.endRecording()
+      ..debugCreator = child;
+    child.append(pictureLayer);
+    return pictureLayer;
+  }
+
+  List<PictureLayer> _processConflictingPhysicalLayers(PhysicalModelLayer predecessor, PhysicalModelLayer child) {
+    FlutterError.reportError(FlutterErrorDetails(
+      exception: FlutterError('Painting order is out of order with respect to elevation.'),
```

Comment:
This could be more informative. Can we give any call to action to resolve the issue? Where can a developer learn more?

Reply:
The full error ends up looking like this: 

```
I/flutter (15065): ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (15065): The following assertion was thrown during compositing:
I/flutter (15065): Painting order is out of order with respect to elevation.
I/flutter (15065): 
I/flutter (15065): Attempted to composite layer:
I/flutter (15065): PhysicalModelLayer#c1c68(creator: PhysicalModel ← AnimatedPhysicalModel ← Material ← ConstrainedBox
I/flutter (15065): ← Container ← Stack ← Column ← Directionality ← [root], elevation: 1.0, color: MaterialColor(primary
I/flutter (15065): value: Color(0xff2196f3)))
I/flutter (15065): after layer:
I/flutter (15065): PhysicalModelLayer#d7ff6(creator: PhysicalModel ← AnimatedPhysicalModel ← Material ← ConstrainedBox
I/flutter (15065): ← Container ← Stack ← Column ← Directionality ← [root], elevation: 1.2, color: MaterialColor(primary
I/flutter (15065): value: Color(0xff4caf50)))
I/flutter (15065): which occupies the same area at a higher elevation.
I/flutter (15065): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

Although unfortunately only for the first one.  Is there some way you think I could highlight that this will be added with more information elsewhere?

Reply:
Maybe we need a doc page on flutter.dev going over this concept?

Reply:
Linking to somewhere with more information would probable be good. For now maybe just link back to the docs of debugCheckElevationsEnabled which have a nice example?